File: loadbalancer.rb
require "socket"
ld=TCPSocket.open("192.168.23.129",1000)

while true
c=ld.accept

rhost=c.peeraddr.last
puts "#{rhost} connect to loadbalancer"

client=c.gets
puts "message from client: #{client}"

if client=="server 1"
server1=TCPSocket.open("192.168.32.128",2000)
server1.write(client)
server1.close

elsif client=="server 2"
server2=TCPSocket.open("192.168.32.129",3000)
server2.write(client)
server2.close

else
puts "ERROR!!"

end
end

File: loadbalancer.java
import java.net.*;

public class loadbalancer{
		public static void main(String []args)throws IOException{
		ServerSocket s=new ServerSocket(1000);
		while(true){
			Socket connection=s.accept();
			BufferedReader message=new BufferedReader(new InputStreamReader(connection.getInputStream()));
			String csentence=message.readLine();

			if(csentence.contains("server 1")){
			Socket s1=new Socket("192.168.32.128",2000);
			DataOutputStream server1=new DataOutputStream(s1.getOutputStream());
			server1.writeBytes(csentence+"\n");
			s1.close();
			}

			else if(csentence.contains("server 2")){
			Socket s2=new Socket("192.168.32.129",3000);
			DataOutputStream server2=new DataOutputStream(s2.getOutputStream());
			server2.writeBytes(ctsentence+"\n");
			s2.close();
			}

			else{System.out.println(ERROR!!);}
			}
		}
}